### PR TITLE
msvc: Support Microsoft Build Tools 2017

### PR DESCRIPTION
**Pull request by hjmallon**
Microsoft Build Tools 2017 is a distribution of the Visual Studio compiler without the UI. It doesn't have devenv but cmake still works using msbuild. With the 2017 version no ENV variables are installed to find vcvarsall but we can use the path to MSBuild to find it (like we can with Devenv in Visual Studio proper).

I was able to reproduce the problem using:
`TOOLCHAIN=vs-15-2017-win64-cxx17 PROJECT_DIR=examples/ZLIB/ ./jenkins.py`

Possibly wording on the devenv wiki page needs changing to add reference to MSBuild?

This issue was mentioned by me here: https://github.com/ruslo/hunter/issues/751#issuecomment-460349156

### CODE REVIEW

**Comment by ruslo** [comment]
> Possibly wording on the devenv wiki page needs changing to add reference to MSBuild?

Feel free to add any useful info there.

Released: https://github.com/ruslo/hunter/releases/tag/v0.23.96

File: cmake/modules/hunter_setup_msvc.cmake [hunk 1]
```diff
@@ -115,32 +115,45 @@ macro(hunter_setup_msvc)
...
+        if(NOT is_devenv_empty AND IS_ABSOLUTE "${CMAKE_VS_DEVENV_COMMAND}")
+          get_filename_component(_hunter_vcvarsall_path
+              "${CMAKE_VS_DEVENV_COMMAND}" DIRECTORY
           )
-        endif()
-        if(NOT IS_ABSOLUTE "${CMAKE_VS_DEVENV_COMMAND}")
+          set(_hunter_vcvarsall_path
+              "${_hunter_vcvarsall_path}/../../VC/Auxiliary/Build"
+          )
+        elseif(is_msbuild_empty AND IS_ABSOLUTE "${CMAKE_VS_MSBUILD_COMMAND}")
```
Comment (ruslo) [inline cmake/modules/hunter_setup_msvc.cmake]: `NOT` is missing? It should be
```cmake
elseif(NOT is_msbuild_empty AND IS_ABSOLUTE "${CMAKE_VS_MSBUILD_COMMAND}")
```
?

Comment (hjmallon) [reply cmake/modules/hunter_setup_msvc.cmake]: Yes I think it should match the Devenv case. Sorry about that.

Comment (ruslo) [reply cmake/modules/hunter_setup_msvc.cmake]: Fix: https://github.com/ruslo/hunter/commit/556dcfea6b4b8051404de4eaac897ef5086c96e0

Released: https://github.com/ruslo/hunter/releases/tag/v0.23.110

> Yes I think it should match the Devenv case. Sorry about that.

> we can use the path to MSBuild to find it

Well, I was thinking it's exactly the branch that do the fix for you. Have you tested it?

`if(NOT is_devenv_empty AND IS_ABSOLUTE "${CMAKE_VS_DEVENV_COMMAND}")` branch used on my local machine and in AppVeyor testing, that's why bug wasn't caught.

Comment (hjmallon) [reply cmake/modules/hunter_setup_msvc.cmake]: You are right. I negated the check to check the error message case, then accidentally pushed the result. I have sent you a beer via PayPal by way of apology.

**Pull request merged**